client/state/analytics/README.md

```diff
@@ -84,6 +84,30 @@ const mapDispatchToProps = {
 `loadTrackingTool :: String -> String -> Object`<br />
 `loadTrackingTool ( trackingTool )`: Creates an action to load a tracking tool that gets attached to the DOM on component mount. Pass tracking tool reference through to middleware and manage loading scripts there.
 
+### Property Enhancers
+
+The above Redux actions denote the analytics data object to be sent to Tracks. Properties to be reported often need to be passed to the actions as arguments. There are times when it would be convenient to enhance the analytics data object with universal properties that are applicable to all sorts of tracking calls. This is when Enhancers come in handy. To enhance any analytics Redux action with additional property:
+
+```
+const recorder = withEnhancer( recordPageView, [ enhancer1, enhancer2, enhancer3,... ] );
+```
+
+#### [enhanceWithSiteType](https://github.com/Automattic/wp-calypso/tree/06b16874d4e833b62e3ec0a53290650711452fe1/client/state/analytics/actions/enhance-with-site-type.js)
```

Comment:
@tyxla Recommended pointing to a [specific commit instead of trunk](https://github.com/Automattic/wp-calypso/pull/66731#discussion_r955744372) to handle cases where the file has been changed and the reference is no longer valid. That said, given that the hash will be changed when the PR gets merged, I'll point to `trunk`. If the file is changed, then I suppose the README would have to be updated anyways, so we shouldn't need to worry about invalid references by referencing `trunk`

Reply:
I missed that comment yesterday, but that I do understand that reasoning. 

Still, I suppose all links under the `enhanceWithSiteMainProduct` probably should reference `trunk` though. Because the documentation refers to functionality implemented in this PR, we don't want to link older versions of those files.

I guess the links with line references (`#L47`) are extra sensitive. But you could either open a new PR after this one has been merged to make those links point to a commit hash, or skip the line references altogether to make the URL's less sensitive.

Reply:
I did recommend this approach as it always leads to the same code. However, we should use a commit that exists in `trunk`, not a commit from a different branch. Does that make sense?

Reply:
> we should use a commit that exists in trunk

There are links linking to a file that only exists in this branch, so I can't really supply the commit here. I'll skip those references in this PR, and do another PR for the README.

Reply:
Makes sense, that's the only way when it's about referencing code that we're introducing in this PR.